engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ModuleScope.java

```diff
@@ -32,22 +31,16 @@ public final class ModuleScope implements TruffleObject {
    * Creates a new object of this class.
    *
    * @param module the module related to the newly created scope.
-   * @param context the current langauge context
    */
-  public ModuleScope(Module module, EnsoContext context) {
+  public ModuleScope(Module module) {
     this.polyglotSymbols = new HashMap<>();
     this.types = new HashMap<>();
     this.methods = new HashMap<>();
     this.conversions = new HashMap<>();
     this.imports = new HashSet<>();
     this.exports = new HashSet<>();
     this.module = module;
-    this.associatedType =
-        Type.createSingleton(
-            module.getName().item(),
-            this,
-            context == null ? null : context.getBuiltins().any(),
-            false);
+    this.associatedType = Type.createSingleton(module.getName().item(), this, null, false);
```

Comment:
Not sure about that. Isn't it breaking the type hierarchy? That Any is a supertype of all types

Reply:
The type hierarchy remains unchanged. This PR:
- trades construction time for time spent in `getSupertype()`
- ensures only builtins (`Any` and `Error` currently) can have no parent
- all other types either specify their parent or get `Any` in `getSupertype()`